scripts/template-updater.ts

```diff
@@ -0,0 +1,197 @@
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+import { err, JSONable, log } from './lib-concourse';
+
+function readFile(filePath: string): string {
+    log(`reading content from file: ${filePath}`);
+    if (fs.existsSync(filePath)) {
+        const stat = fs.statSync(filePath);
+        if (stat.isFile()) {
+            log(`file found: ${filePath}`);
+            const buffer = fs.readFileSync(filePath);
+            return buffer.toString();
+        } else {
+            log(`not a file: ${filePath}`);
+        }
+    } else {
+        log(`file not found: ${filePath}`);
+    }
+    log(`unable to read content from file: ${filePath}`);
+    return null;
+}
+
+function readJSONTemplate(filePath: string): JSONable {
+    try {
+        const content = readFile(filePath);
+        return content !== null && JSON.parse(content);
+    } catch (error) {
+        err('error in parsing into JSON object: ', JSON.stringify(error));
+    }
+    return null;
+}
+
+function writeFile(filePath: string, content: string): void {
+    log(`writing content into file: ${filePath}`);
+    if (fs.existsSync(filePath)) {
+        const stat = fs.statSync(filePath);
+        if (stat.isFile()) {
+            log(`file found: ${filePath}`);
+            fs.writeFileSync(filePath, content);
+            log('write file complete');
+            return;
+        } else {
+            log(`not a file: ${filePath}`);
+        }
+    } else {
+        log(`file not found: ${filePath}`);
+    }
+    log(`unable to write content to file: ${filePath}`);
+    return;
+}
```

Comment:
This seems like you are just re-creating all the code that is inside `fs.writeFileSync`

Reply:
yes. it gives more log output, and prevents from throwing IO level errors